app/services/fees/subscription_service.rb

```diff
@@ -218,12 +218,14 @@ def single_day_price(target_subscription, optional_from_date: nil)
       )
     end
 
-    def compute_to_date(target_subscription, base_date)
+    def compute_old_to_date(old_subscription, base_date)
       return base_date if plan.pay_in_arrear?
 
+      old_base_date = date_service(old_subscription).to_date
+
       # NOTE: when plan is pay in advance, the_to date should be the
       #       end of the actual period
-      date_service(target_subscription).next_end_of_period(base_date)
+      date_service(old_subscription).next_end_of_period(old_base_date)
```

Comment:
Shouldn't we just call: `date_service(old_subscription).next_end_of_period(boundaries.timestamp)` here?

Reply:
it can do the job but I'm not sure, I may try this